catalog/ui/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/util/EndpointUtil.java

```diff
@@ -135,6 +155,91 @@ public Metacard getMetacard(String id)
         return result.getMetacard();
     }
 
+    /**
+     * Returns all associated metacards given two relatable attributes.
+     *
+     * @param primaryFilterAttribute   The primary attribute to search against
+     * @param secondaryFilterAttribute The secondary attribute to search against
+     * @param primaryAttributeValue    A primary attribute value
+     * @param secondaryAttributeValue  A secondary attribute value
+     * @return A set of metacards, filtered by the given attributes and it's values
+     * @throws UnsupportedQueryException  Error
+     * @throws SourceUnavailableException Error
+     * @throws FederationException        Error
+     */
+    public List<Metacard> getAssociatedMetacardsByTwoAttributes(String primaryFilterAttribute,
+            String secondaryFilterAttribute,
+            String primaryAttributeValue,
+            String secondaryAttributeValue)
+            throws UnsupportedQueryException, SourceUnavailableException, FederationException {
```

Comment:
Throwing the CatalogFramework exceptions is consistent with the other methods in EndpointUtil that deal with CatalogFramework exceptions.